### PR TITLE
feat: plumb from through UTXOs `buildThorTrade`

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

Does what it says on the box - plumbs `from` through `buildThorTrade` for UTXOs - Cosmos SDK was already handled, and EVM chains is the only one which will still need to be tackled and require quite a bit more lift.

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

## Risk

None, multi-hop isolated

### Engineering

- Implementation looks sane

### Operations

Nothing to see here

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #4826** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4826" target="_blank">Graphite</a> 
    * **PR #4827** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4827" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/4827?utm_source=stack-comment).

File: src/lib/swapper/swappers/ThorchainSwapper/buildThorTrade/buildThorTrade.ts [hunk 1]
```diff
@@ -113,13 +113,17 @@ export const buildTrade = async (
     if (maybeThorTxInfo.isErr()) return Err(maybeThorTxInfo.unwrapErr())
     const { vault, opReturnData } = maybeThorTxInfo.unwrap()
 
-    // TODO(gomes): it is not - plumb from through
```
Comment (gomesalexandre) [inline src/lib/swapper/swappers/ThorchainSwapper/buildThorTrade/buildThorTrade.ts:116]: Actually is to get from, but doesn't need to be passed down - that's why we have `buildSignTx`

File: src/lib/swapper/swappers/ThorchainSwapper/buildThorTrade/buildThorTrade.ts [hunk 2]
```diff
@@ -112,12 +113,17 @@ export const buildTrade = async (
     if (maybeThorTxInfo.isErr()) return Err(maybeThorTxInfo.unwrapErr())
     const { vault, opReturnData } = maybeThorTxInfo.unwrap()
 
-    // TODO(gomes): update to use from instead of wallet
-    const buildTxResponse = await (
-      sellAdapter as unknown as UtxoBaseAdapter<ThorUtxoSupportedChainId>
-    ).buildSendTransaction({
-      value: sellAmountCryptoBaseUnit,
+    if (!wallet) throw new Error('wallet is required to make a trade Tx')
```
Comment (kaladinlight) [inline src/lib/swapper/swappers/ThorchainSwapper/buildThorTrade/buildThorTrade.ts:116]: wallet is required prop, reason for this check?

Comment (gomesalexandre) [reply src/lib/swapper/swappers/ThorchainSwapper/buildThorTrade/buildThorTrade.ts]: Nice catch, probably a leftover of wallet being optional at some point in the initial PR - removed in https://github.com/shapeshift/web/pull/4827/commits/043646bde09988f733ea0ec760d0c830a440a6d2

**Review by kaladinlight — DISMISSED** [review]

**Comment by gomesalexandre** [comment]
[Graphite](https://app.graphite.dev/github/pr/shapeshift/web/4827) rebased this pull request as part of a merge.

**Comment by gomesalexandre** [comment]
[Graphite](https://app.graphite.dev/github/pr/shapeshift/web/4827) couldn't merge this PR because it was not satisfying all requirements.

**Review by kaladinlight — APPROVED** [review]

**Pull request merged**